Makefile

```diff
@@ -5,21 +5,21 @@ PROGRAMS_DIR=./programs
 ZKSOLC_YUL_FLAGS=--asm --bin --yul --overwrite
 ZKSOLC_ASM_FLAGS=--zkasm --bin --overwrite
 
-YUL_PROGRAMS = $(wildcard $(PROGRAMS_DIR)/*.yul)
-ASM_PROGRAMS = $(wildcard $(PROGRAMS_DIR)/*.zasm)
-ARTIFACTS_YUL = $(patsubst $(PROGRAMS_DIR)/%.yul, $(ARTIFACTS_DIR)/%.artifacts.yul, $(YUL_PROGRAMS))
-ARTIFACTS_ASM = $(patsubst $(PROGRAMS_DIR)/%.zasm, $(ARTIFACTS_DIR)/%.artifacts.zasm, $(ASM_PROGRAMS))
+YUL_PROGRAMS := $(shell find $(PROGRAMS_DIR) -type f -name "*.yul")
```

Comment:
Why was this change needed?

Reply:
Not really needed, it's only to avoid eventually having to many similarly named tests on the same directory
![image](https://github.com/lambdaclass/era_vm/assets/62400508/fcc0d710-3d67-4ae4-969e-d340ece42b24)

Reply:
Makes sense to me, let's just be 100% sure that it does not bring any unwanted problems when running tests and such.